patina_dxe_core/src/config_tables/memory_attributes_table.rs

```diff
@@ -142,22 +143,32 @@ pub fn core_install_memory_attributes_table() {
         POST_RTB.call_once(|| {});
     }
 
-    // get the GCD memory map descriptors and filter out the non-runtime sections
-    let desc_list = match get_memory_map_descriptors(true) {
-        Ok(descriptors) => descriptors,
+    // Reserve a buffer for the memory map descriptors
+    let descriptor_count = GCD.memory_descriptor_count();
+    let mut desc_buffer = Vec::with_capacity(descriptor_count);
+    desc_buffer.resize(
+        descriptor_count,
+        efi::MemoryDescriptor { r#type: 0, physical_start: 0, virtual_start: 0, number_of_pages: 0, attribute: 0 },
+    );
+
+    let actual_count = match GCD.populate_efi_memory_map(&mut desc_buffer, true) {
```

Comment:
nit: i would use a let Ok statement here since you dont' care about the actual error

```rust
let Ok(actual_count) = GCD.populate_efi_memory_map(&mut desc_buffer, true) else {
  log::error!("Failed to populate memory map descriptors.");
  return;
}
```